Export heron-spi jar

ajorgensen opened the pull request:
The heron-spi jar is needed for developing custom schedulers, packing
algorithms, etc. We will want to upload this jar to maven central along
with the api and storm compatibility layer.

Related to: https://github.com/twitter/heron/issues/935

kramasamy commented on heron/spi/src/java/BUILD:
```diff
@@ -146,3 +146,19 @@ java_library(
+    srcs = [":spi-unshaded_deploy.jar"],
+    outs = ["heron-spi.jar"],
+    cmd  = "cp $< $@",
+)
```
This might not be enough - we might have to package this jar along with storm and api jars as well - so that we can publish to maven and goes into the install scripts.

ajorgensen replied:
My thought is that this would live along side the api and storm jars and can be added as dependency the same way as those two. I think generally anything that coding against the spi interfaces will be expected to be run from the heron-cli or something like that right?

kramasamy replied:
We need to get this jar into heron-api install scripts - so that when a user installs he/she gets everything.

ajorgensen replied:
Ah sorry I misunderstood. I can look into adding it there as well.

On Friday, June 17, 2016, Karthik Ramasamy <notifications@github.com> wrote:

> In heron/spi/src/java/BUILD
> <https://github.com/twitter/heron/pull/937#discussion_r67466587>:
>
> > +
> > +java_binary(
> > +    name = "spi-unshaded",
> > +    srcs = glob([
> > +        "**/spi/utils/**/*.java",
> > +        "**/spi/common/**/*.java",
> > +    ]),
> > +    deps = utils_deps_files,
> > +)
> > +
> > +genrule(
> > +    name = "heron-spi-jar",
> > +    srcs = [":spi-unshaded_deploy.jar"],
> > +    outs = ["heron-spi.jar"],
> > +    cmd  = "cp $< $@",
> > +)
>
> We need to get this jar into heron-api install scripts - so that when a
> user installs he/she gets everything.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/twitter/heron/pull/937/files/b4087fad325028efed0c3a317bd786894a4f1bd2#r67466587>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAfHUyQ3gTVlLHq9HCXpDGZTr94GEPD_ks5qMkNXgaJpZM4I3zHj>
> .
>


-- 
----
Andrew Jorgensen
@ajorgensen

ajorgensen commented:
@kramasamy I added the spi jar to the heron-api-install.sh flow and tested it out locally. I added the spi jar to the list returned by `heron_api_lib_files`, let me know if thats not the appropriate place to put it. It seemed logical to group them together and as far as I could tell the only place that method was used was in the generation of `heron-api.tar.gz` and the install script.

kramasamy commented:
👍

kramasamy commented:
Alternatively, if you check in whatever you have, I add a separate package called 'spi' in the next iteration.
